### PR TITLE
Separated tablet scan functions from TabletClientService into a new Thrift service

### PR DESCRIPTION
**Pull request by dlmarion**
related to #2640

### CODE REVIEW

**Comment by dlmarion** [comment]
IT Test run had 2 failures. SSLIT.adminStop timed out after 3 minutes and LateLastContactIT timed out. All others passed at https://github.com/dlmarion/accumulo/runs/6099910179

**Comment by DomGarguilo** [comment]
> SSLIT.adminStop timed out after 3 minutes 

The timeouts for this IT were recently adjusted (see #2631) so it might just be that the timeout needs to be increased for this test

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+  public ThriftScanClientHandler(TabletServer server, WriteTracker writeTracker) {
+    this.server = server;
+    this.context = server.getContext();
+    this.writeTracker = writeTracker;
+    this.security = AuditedSecurityOperation.getInstance(context);
+    MAX_TIME_TO_WAIT_FOR_SCAN_RESULT_MILLIS = server.getContext().getConfiguration()
+        .getTimeInMillis(Property.TSERV_SCAN_RESULTS_MAX_TIMEOUT);
+  }
+
+  private void checkPermission(TCredentials credentials, String lock, final String request)
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java:111]: Is this method identical to what is in TabletClientHandler?  If so may be good to share the code.  Maybe once class could have a static method that they both call.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+    if (extent.isRootTablet() || extent.isMeta()) {
+      // dispatcher is only for user tables
+      return null;
+    }
+
+    return context.getTableConfiguration(extent.tableId()).getScanDispatcher();
+  }
+
+  @Override
+  public InitialScan startScan(TInfo tinfo, TCredentials credentials, TKeyExtent textent,
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java:185]: Were all of these overridden methods copied exactly from TabletClientHandler?  Wondering if there any changes to look at.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java:111]: Yes, should be a copy. I can consolidate.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java (hunk 2 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java:185]: Should be an exact copy, there were no changes to the Thrift method signatures, just moving them. IIRC, I copied the code and pasted it into a new class.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java]: Addressed in d4b1a76

**Review by keith-turner — APPROVED** [review]

**Pull request merged**